packages/wdio-browserstack-service/src/service.ts

```diff
@@ -1,12 +1,72 @@
 import logger from '@wdio/logger'
-import got from 'got'
+import got, { GotRequestFunction } from 'got'
+import WebDriver from 'webdriver'
 
 import { getBrowserDescription, getBrowserCapabilities, isBrowserstackCapability } from './util'
 
 const log = logger('@wdio/browserstack-service')
 
+type Capabilities = WebDriver.Capabilities & WebdriverIO.MultiRemoteCapabilities;
+
+type Browser = WebdriverIO.BrowserObject & WebdriverIO.MultiRemoteBrowserObject;
+
+type Config = WebdriverIO.Config & { cucumberOpts?: { strict: boolean } };
+
+type Suite = {
+    title: string;
+}
+
+type Test = {
+    fullName: string,
+    parent: string,
+    title: string
+};
+
+type Context = any;
+
+type Specs = any;
+
+type Feature = {
+    document: {
+        feature: {
+            name: string;
+        }
+    }
+};
+
+type Results = {
+    error: Error;
+    passed: boolean
+    exception: any;
+    status: string;
+};
+
+type Pickle = {
+    name: string;
+}
+
+type MultiRemoteAction = (sessionId: string, browserName?: string) => ReturnType<GotRequestFunction> | Promise<void>;
+
+type SessionResponse = {
+    // eslint-disable-next-line camelcase
+    automation_session: {
```

Comment:
@christian-bromann I have created types here just to simulate what would be needed to solve all the errors in this file, however I'm not sure where they should come from, I'd need some help here! :)